src/features/contact/contactverzoek/overzicht/service.ts

```diff
@@ -1,95 +1,126 @@
-  url.searchParams.set("pageSize", "10");
+  return fetchLoggedIn(url)
+    .then(throwIfNotOk)
+    .then(parseJson)
+    .then(async (j) => {
+      if (!Array.isArray(j?.results)) {
+        throw new Error("Expected array: " + JSON.stringify(j));
+      }
 
-  url.searchParams.set(
-    "data_attrs",
-    `betrokkene__digitaleAdressen__icontains__${query}`,
-  ); //todo: kan je in 1 call op email OF telefoonnr zoeken? anders de interface aanpassen zodat duidelijk is dat je maar naar een van beide kan zoeken (bv samenvoegen in 1 zoekveld!)
+      if (!j.next) return j.results;
+      const nextResults = await searchRecursive(urlStr, page + 1);
+      return [...j.results, ...nextResults];
+    });
+}
 
-  return url.toString();
+async function searchOk2Recursive(
+  adres: string,
+  page = 1,
+): Promise<DigitaalAdresExpandedApiViewModel[]> {
+  const paginated = await searchDigitaleAdressen({
+    adres,
+    page,
+    expand: [DigitaleAdressenExpand.verstrektDoorBetrokkene],
+  });
+  if (!paginated.next) return paginated.page;
+  return [...paginated.page, ...(await searchOk2Recursive(adres, page + 1))];
 }
 
-function searchRecursive(urlStr: string, page = 1): Promise<any[]> {
-  //recursive alle pagina's ophalen.
-  //set de page param van de huidige op te halen pagina
-  const url = new URL(urlStr);
-  url.searchParams.set("page", page.toString());
+export async function search(
+  query: string,
+  gebruikKlantInteractiesApi: boolean,
+): Promise<Contactverzoek[]> {
+  // OK2
+  if (gebruikKlantInteractiesApi) {
+    const adressen = await searchOk2Recursive(query);
 
-  return (
-    fetchLoggedIn(url)
-      .then(throwIfNotOk)
-      .then(parseJson)
-      //todo: parsePagination toevoegen??
-      .then(async (j) => {
-        if (!Array.isArray(j?.results)) {
-          throw new Error("expected array: " + JSON.stringify(j));
-        }
+    const betrokkenen = adressen
+      .map((x) => x?._expand?.verstrektDoorBetrokkene)
+      .filter(Boolean) as Betrokkene[];
 
-        const result: any[] = [];
-        j.results.forEach((k: any) => {
-          result.push(k);
-        });
+    const uniqueBetrokkenen = new Map<string, Betrokkene>(
+      betrokkenen.map((betrokkene) => [betrokkene.uuid, betrokkene]),
+    );
 
-        if (!j.next) return result;
-        return await searchRecursive(urlStr, page + 1).then((next) => [
-          ...result,
-          ...next,
-        ]);
-      })
-  );
-}
+    return (
+      enrichBetrokkeneWithKlantContact(
+        [...uniqueBetrokkenen.values()],
+        [KlantContactExpand.leiddeTotInterneTaken],
+      )
+        .then(filterOutContactmomenten)
+        .then(enrichBetrokkeneWithDigitaleAdressen)
+        .then(enrichInterneTakenWithActoren)
+        .then(mapToContactverzoekViewModel)
+        // Filter voor OK2: alleen resultaten zonder 'klant'
+        .then((r) => r.filter((x) => !x.record.data.betrokkene.klant))
```

Comment:
[[[https://github.com/Klantinteractie-Servicesysteem/KISS-frontend/blob/247fd85148dce11bde168a3faef0be6e9a4bbij anoniemen contactverzoeken.290c/src/features/contact/contactmoment/ContactmomentAfhandeling.vue#L518 er wordt ook een betrokkene aangemaakt bij anoniemen contactverzoeken.](https://github.com/Klantinteractie-Servicesysteem/KISS-frontend/blob/247fd85148dce11bde168a3faef0be6e9a4b290c/src/features/contact/contactmoment/ContactmomentAfhandeling.vue#L526)](https://github.com/Klantinteractie-Servicesysteem/KISS-frontend/blob/247fd85148dce11bde168a3faef0be6e9a4b290c/src/features/contact/contactmoment/ContactmomentAfhandeling.vue#L526)](https://github.com/Klantinteractie-Servicesysteem/KISS-frontend/blob/247fd85148dce11bde168a3faef0be6e9a4b290c/src/features/contact/contactmoment/ContactmomentAfhandeling.vue#L526)

Gaat dat filteren zo goed? Er wordt ook een betrokkene aangemaakt bij een contactverzoek zonder klant.

Reply:
ja, ik map de url van de partij hierop. als er geen partij is, is klant leeg